multicluster/service-mirror/cluster_watcher.go

```diff
@@ -1110,33 +1110,43 @@ func (rcsw *RemoteClusterServiceWatcher) handleCreateOrUpdateEndpoints(
 			},
 		}
 	}
-	return rcsw.createOrUpdateEndpoints(ctx, ep)
+	return rcsw.updateMirrorEndpoints(ctx, ep)
 }
 
-// createOrUpdateEndpoints will create or update endpoints based off gateway
-// liveness. If the gateway is not alive, then the addresses in each subset
-// will be set to not ready; these will moved back to ready once the gateway
-// is alive and an update is received.
-func (rcsw *RemoteClusterServiceWatcher) createOrUpdateEndpoints(ctx context.Context, endpoints *corev1.Endpoints) error {
+// createMirrorEndpoints will create endpoints based off gateway liveness. If
+// the gateway is not alive, then the addresses in each subset will be set to
+// not ready.
+func (rcsw *RemoteClusterServiceWatcher) createMirrorEndpoints(ctx context.Context, endpoints *corev1.Endpoints) error {
 	if !rcsw.gatewayAlive {
 		rcsw.log.Warnf("gateway for %s/%s does not have ready addresses; setting addresses to not ready", endpoints.Namespace, endpoints.Name)
 		for i := range endpoints.Subsets {
 			endpoints.Subsets[i].NotReadyAddresses = append(endpoints.Subsets[i].NotReadyAddresses, endpoints.Subsets[i].Addresses...)
 			endpoints.Subsets[i].Addresses = nil
 		}
 	}
-	_, err := rcsw.localAPIClient.Client.CoreV1().Endpoints(endpoints.Namespace).Get(ctx, endpoints.Name, metav1.GetOptions{})
+	_, err := rcsw.localAPIClient.Client.CoreV1().Endpoints(endpoints.Namespace).Create(ctx, endpoints, metav1.CreateOptions{})
 	if err != nil {
-		if !kerrors.IsNotFound(err) {
-			return fmt.Errorf("Failed to get endpoints for %s/%s: %w", endpoints.Namespace, endpoints.Name, err)
-		}
-		_, err := rcsw.localAPIClient.Client.CoreV1().Endpoints(endpoints.Namespace).Create(ctx, endpoints, metav1.CreateOptions{})
-		if err != nil {
-			return fmt.Errorf("Failed to create endpoints for %s/%s: %w", endpoints.Namespace, endpoints.Name, err)
+		return fmt.Errorf("Failed to create mirror endpoints for %s/%s: %w", endpoints.Namespace, endpoints.Name, err)
+	}
+	return nil
+}
+
+// updateMirrorEndpoints will update endpoints based off gateway liveness. If
+// the gateway is not alive, then the addresses in each subset will be set to
+// not ready. Future calls to updateMirrorEndpoints can set the addresses back
+// to ready if the gateway is alive.
+func (rcsw *RemoteClusterServiceWatcher) updateMirrorEndpoints(ctx context.Context, endpoints *corev1.Endpoints) error {
+	if !rcsw.gatewayAlive {
```

Comment:
is it worth factoring this block into a `rcsw.UpdateReadiness(endpoints)` function which can be shared between updateMirrorEndpoints and createMirrorEndpoints?